processing/data_collection/gazette/spiders/pr_curitiba.py

```diff
@@ -0,0 +1,120 @@
+            callback=self.parse_month
+        )
+
+    def parse_month(self, response):
+        #Count how many pages and iterate
+        page_count = len(response.css(".grid_Pager:nth-child(1) table td").extract())
+        month = response.meta["month"]
+        #The first page of pagination cannot be accessed by page number
+        yield scrapy.FormRequest.from_response(
+            response,
+            formdata={
+                '__EVENTTARGET': 'ctl00$cphMasterPrincipal$TabContainer1',
+                'ctl00_cphMasterPrincipal_TabContalegacyDealPooliner1_ClientState' : '{{"ActiveTabIndex":{},"TabState":[true,true,true,true,true,true,true,true,true,true,true,true]}}'.format(month),
+                '__EVENTARGUMENT': 'activeTabChanged:{}'.format(month),
+            },
+            callback=self.parse_page,
+        )
+        for page_number in range(2,page_count + 1):
+            yield scrapy.FormRequest.from_response(
+                response,
+                formdata={
+                    '__EVENTARGUMENT' : 'Page${}'.format(page_number),
+                    '__EVENTTARGET' : 'ctl00$cphMasterPrincipal$gdvGrid2'
+                },
+                callback=self.parse_page,
+            )
+    def parse_page(self, response):
+        row = response.css(".grid_Row")
+        numbers = row.css("td:nth-child(1) span ::text").extract()
+        pdf_dates = row.css("td:nth-child(2) span ::text").extract()
+        ids = row.css("td:nth-child(3) a ::attr(data-teste)").extract()
+        for i in range(len(numbers)):
+            number = numbers[i]
+            pdf_date = pdf_dates[i]
+            id = ids[i]
+            parsed_date = parse(f'{pdf_date}', languages=['pt']).date()
+            if id == '0':
+                yield scrapy.FormRequest.from_response(
+                    response,
+                    headers = {'user-agent': 'Mozilla/5.0'},
+                    formdata = {
+                        '__LASTFOCUS': '',
+                        '__EVENTTARGET': 'ctl00$cphMasterPrincipal$gdvGrid2$ctl{num:02d}$lnkVisualizar'.format(num=(i+3)),
+                        '__EVENTARGUMENT': '',
+                        '__ASYNCPOST': 'true'
+                    },
+                    callback=self.scrap_not_extra_edition,
+                    meta={"parsed_date": parsed_date}
+                )
+            else:
+                yield Gazette(
+                    date = parsed_date,
+                    file_urls=["http://legisladocexterno.curitiba.pr.gov.br/DiarioSuplementoConsultaExterna_Download.aspx?id={}".format(id)],
+                    is_extra_edition= True,
+                    municipality_id=self.MUNICIPALITY_ID,
+                    power='executive_legislature',
+                    scraped_at=dt.datetime.utcnow()
+                )
+
+    def scrap_not_extra_edition(self, response):
+        parsed_date = response.meta['parsed_date']
+        id = re.findall(r'Id=(\d+)', response.text )[0]
```

Comment:
Don't do that! You have no idea if your regex will match something. You'll get a big `IndexError` when it happens!
You can use the _re_ method in _response_. See https://doc.scrapy.org/en/latest/topics/selectors.html#using-selectors-with-regular-expressions

Reply:
The response for this request is not a valid html, I believe I can't call re from response because I can't get a valid return from any selector, correct?

About raising exceptions, if there is no "Id=" I wouldn't want to ignore this error, because it would mean that the website structure changed. I agree  that IndexError is a really bad exception to raise when this happens.
Do you have any suggestion on how should I deal with it in a way that it would be clear that something went wrong? Should I throw a more explanatory exception or is there another way to log this problem?

Reply:
@rennerocha @cuducos @Irio We don't know what must be done when an unexpected return happen. Should we fail silently or throw an exception? We guess that is the last change before the PR can be accepted.

Reply:
Always try to handle the problem. But this is a avoidable problem.

In this case I didn't find where in the page this `Id=(\d+)` is.
For sure it is a string, so you can use the `re_first` method: `response.selector.re_first('Id=(\d+)')`.

Reply:
If the page changes in future (and I hope so, because dealing with these _VIEWSTATES are terrible), we will notice monitoring it and realizing that no items were scraped.